src/main/webapp/data-visualization.js

```diff
@@ -312,16 +318,34 @@ function setStyle(isCountyQuery) {
   }
 }
 
+// Changes the color of the current visualizations on the page.
+function changeColor(colorParam) {
+  const color = colorParam.value;
+  localStorage.setItem('color', color);
+  // Get variables out of cache
+  const cacheMapsData = JSON.parse(localStorage.getItem('mapsData'));
+  const cacheGeoData = JSON.parse(localStorage.getItem('geoData'));
+  const cacheAmCharts = JSON.parse(localStorage.getItem('amChartsData'));
+  const cacheLocation = localStorage.getItem('location');
+  const cacheDescription = localStorage.getItem('description');
+  // map is undefined on a state query, so check to be sure that 
+  // it is undefined before calling displayCountyGeoJson.
+  if (typeof cacheMapsData !== 'undefined') {
```

Comment:
This would only be undefined at the very beginning when you are showing the full US map and so not saving this value to localStorage. But once you make a county query that value will be stored here and since you never delete this value it will have the last save values. so maybe you have actually have to pass in whether this is a county query or not as a parameter and make this a conditional on that.

Reply:
location is actually "state" for a state query and "01", "02" etc... for a county query where the location represents the state number, so I updated the conditional based on when cacheLocation is not "state"